### PR TITLE
Add package py-macs3 and dependencies

### PR DESCRIPTION
**Pull request by snehring**
I'd debated renaming the py-macs2 package to py-macs and adding this to it, but they're separate projects on pypi, so I suppose py-macs3 is fine.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-macs3/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class PyMacs3(PythonPackage):
+    """MACS: Model-based Analysis for ChIP-Seq"""
+
+    homepage = "https://github.com/macs3-project/MACS/"
+    pypi = "MACS3/MACS3-3.0.0b3.tar.gz"
+
+    maintainers("snehring")
+
+    version("3.0.0b3", sha256="caa794d4cfcd7368447eae15878505315dac44c21546e8fecebb3561e9cee362")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-macs3/package.py:18]: Python 3.8+ required

File: var/spack/repos/builtin/packages/py-macs3/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-setuptools@60.0:", type="build")
+    depends_on("py-cython@0.29:0", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-macs3/package.py]: Allegedly this is also required at run-time

File: var/spack/repos/builtin/packages/py-macs3/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    depends_on("py-numpy@1.19:", type=("build", "run"))
+    depends_on("py-cykhash@2.0:2", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-macs3/package.py]:
```suggestion
    depends_on("py-cykhash@2", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-macs3/package.py (hunk 1 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/py-macs3/package.py:18]: got to looking deeper in the setup.py and it actually calls out 3.9+. Consistency!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**